src/main/java/apoc/map/Maps.java

```diff
@@ -178,18 +178,34 @@ public Object get(@Name("map") Map<String,Object> map, @Name("key") String key,
     }
 
     @UserFunction
-    @Description("apoc.map.removeKey(map,key)")
-    public Map<String,Object> removeKey(@Name("map") Map<String,Object> map, @Name("key") String key) {
+    @Description("apoc.map.removeKey(map,key,recursive:false) - remove the key from the map (recursively if recursive is true)")
+    public Map<String,Object> removeKey(@Name("map") Map<String,Object> map, @Name("key") String key, @Name(value="recursive", defaultValue = "false") boolean recursive) {
         Map<String, Object> res = new LinkedHashMap<>(map);
```

Comment:
should we check first if the current map contains the key before creating that new map?
otherwise we can just return the original map